### PR TITLE
refactor: update types in elabQpf to reflect the code

### PR DESCRIPTION
**Pull request by Equilibris**

### CODE REVIEW

**Comment by alexkeizer** [comment]
Can you merge master, now that #21 got merged

File: Qpf/Macro/Comp.lean [hunk 1]
```diff
@@ -154,135 +154,115 @@ def DVec.toExpr {n : Nat} {αs : Q(Fin2 $n → Type u)} (xs : DVec (fun (i : Fin
 
 
 structure ElabQpfResult (u : Level) (arity : Nat) where
-  F : Q(TypeFun.{u,u} $arity)
-  functor : Q(MvFunctor $F)
-  qpf     : Q(@MvQPF _ $F $functor)
+  F : Q(TypeFun.{u,u} $arity)       := by exact q(by infer_instance)
```
Comment (Equilibris) [inline Qpf/Macro/Comp.lean:157]:
```suggestion
  F : Q(TypeFun.{u,u} $arity)
```

File: Qpf/Macro/Comp.lean [hunk 2]
```diff
@@ -154,135 +154,115 @@ def DVec.toExpr {n : Nat} {αs : Q(Fin2 $n → Type u)} (xs : DVec (fun (i : Fin
...
+  functor : Q(MvFunctor $F)         := by exact q(by infer_instance)
+  qpf     : Q(@MvQPF _ $F $functor) := by exact q(by infer_instance)
 deriving Inhabited
 
+def isLiveVar (varIds : Vector FVarId n) (id : FVarId) := varIds.toList.contains id
+
```
Comment (Equilibris) [inline Qpf/Macro/Comp.lean:163]: Fine for now, but let's eventually get `Vector.contains` upstream and just use that

File: Qpf/Macro/Comp.lean [hunk 3]
```diff
@@ -154,135 +154,115 @@ def DVec.toExpr {n : Nat} {αs : Q(Fin2 $n → Type u)} (xs : DVec (fun (i : Fin
...
+    return { F := comp, functor, qpf }
 
-partial def handleArrow (vars : Vector Q(Type u) arity) (target : Q(Type u)) (targetStx : Option Term := none) (normalized := false) : TermElabM (ElabQpfResult u arity) := do
-  match target with
-  | Expr.forallE _ e₁ e₂ .. =>
-    let newTarget ← mkAppM ``MvQPF.Arrow.Arrow #[e₁, e₂]
-    elabQpf vars newTarget targetStx normalized
-  | _ => unreachable!
 
+partial def handleArrow (binderType body : Expr) (vars : Vector FVarId arity) (targetStx : Option Term := none) (normalized := false): TermElabM (ElabQpfResult u arity) := do
```
Comment (Equilibris) [inline Qpf/Macro/Comp.lean:239]:
```suggestion
partial def handleArrow (binderType body : Expr) (vars : Vector FVarId arity) (targetStx : Option Term := none) (normalized := false) : TermElabM (ElabQpfResult u arity) := do
```

File: Qpf/Macro/Comp.lean [hunk 4]
```diff
@@ -154,135 +154,115 @@ def DVec.toExpr {n : Nat} {αs : Q(Fin2 $n → Type u)} (xs : DVec (fun (i : Fin
...
-partial def handleApp (vars : Vector Q(Type u) arity) (target : Q(Type u))  : TermElabM (ElabQpfResult u arity) := do
-  let vars' := vars.toList;
+  pure { F := const, functor := q(Const.MvFunctor), qpf := q(Const.mvqpf)}
 
-  let varIds := vars'.map fun expr => expr.fvarId!
-  let isLiveVar : FVarId → Bool
-    := fun fvarId => (List.indexOf' fvarId varIds).isSome
 
-  let ⟨m, F, args⟩ ← (Comp.parseApp isLiveVar target)
+partial def handleApp (vars : Vector FVarId arity) (target : Q(Type u))  : TermElabM (ElabQpfResult u arity) := do
```
Comment (Equilibris) [inline Qpf/Macro/Comp.lean:186]:
```suggestion
partial def handleApp (vars : Vector FVarId arity) (target : Q(Type u)) : TermElabM (ElabQpfResult u arity) := do
```

**Review by alexkeizer — APPROVED** [review]
LGTM

**Pull request merged**